packages/plugin-expo-eas-sourcemaps/lib/bugsnag-expo-xcode.sh

```diff
@@ -0,0 +1,62 @@
+#/bin/bash
+
+set -o errexit
+
+INFO_PLIST=$BUILT_PRODUCTS_DIR/$INFOPLIST_PATH
+APP_VERSION=$(/usr/libexec/PlistBuddy -c "print :CFBundleShortVersionString" "$INFO_PLIST")
+BUNDLE_VERSION=$(/usr/libexec/PlistBuddy -c "print :CFBundleVersion" "$INFO_PLIST")
+
+API_KEY="$BUGSNAG_API_KEY"
+if [ -z "$API_KEY" ]; then
+  API_KEY=$(/usr/libexec/PlistBuddy -c "print :bugsnag:apiKey" "$INFO_PLIST" || echo)
+fi
+if [ -z "$API_KEY" ]; then
+  echo "No Bugsnag API key detected - add your key to your Info.plist or BUGSNAG_API_KEY environment variable"
+  exit 1
+fi
+
+# in RN 0.64+ the JS bundle is in this location
+BUNDLE_FILE="$CONFIGURATION_BUILD_DIR/main.jsbundle"
+if [ ! -f "$BUNDLE_FILE" ]; then
+  # in RN <0.64 it's in this location
+  BUNDLE_FILE="$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH/main.jsbundle"
+fi
+if [ ! -f "$BUNDLE_FILE" ]; then
+  echo "Skipping source map upload because app bundle could not be found."
+  exit 0
```

Comment:
seems serious, should this fail the script?

Reply:
This is a normal thing to happen if the app has not been bundled - e.g. for debug builds.

Reply:
Ahh, I see. Then this check should probably be first in the script, since if the script shouldn't be run, it also shouldn't fail on a missing API key (and prevent building when it won't be used).

Reply:
This script is basically the same as https://github.com/bugsnag/bugsnag-js/blob/next/packages/react-native/bugsnag-react-native-xcode.sh and there's a desire to keep them as similar as possible.

I would argue that checking the API key still makes sense, in order to alert the developer sooner of incorrect project set-up.

Reply:
I'm not saying to _not_ check the API key, just to do it after the bundle file check, since it doesn't make sense to fail the build for something that won't be used anyway. Don't feel strongly enough beyond that though.